gprofiler/merge.py

```diff
@@ -109,13 +109,19 @@ def _collapse_stack(comm: str, stack: str) -> str:
         # append kernel annotation
         elif "kernel" in dso or "vmlinux" in dso:
             sym += "_[k]"
+        elif inject_dso and dso != "[unknown]":
```

Comment:
Be explicit - if `include_dso_name` was asked, ALWAYS add the DSO name, even if it's unknown